src/cftime/_cftime.pyx

```diff
@@ -1237,6 +1236,32 @@ The default format of the string produced by strftime is controlled by self.form
             format = self.format
         return _strftime(self, format)
 
+    @staticmethod
+    def strptime(datestring, format, calendar='standard', has_year_zero=None):
+        """
+        Return a datetime corresponding to date_string, parsed according to format, 
+        with a specified calendar and year zero convention. 
+        For a complete list of formatting directives, see section
+        'strftime() and strptime() Behavior' in the base Python documentation.
+        """
+        # if possible use python's datetime.strptime to get a python datetime instance
+        # (works for dates in proleptic_gregorian calendar)
+        try:
+            pydatetime = datetime_python.strptime(datestring, format)
+            # remove time zone offset
+            if getattr(pydatetime, 'tzinfo',None) is not None:
+                 pydatetime = pydatetime.replace(tzinfo=None) - pydatetime.utcoffset()
+            # convert the cftime datetime instance
+            return datetime(pydatetime.year, pydatetime.month, pydatetime.day,
+                            pydatetime.hour, pydatetime.minute, pydatetime.second,
+                            pydatetime.microsecond, calendar=calendar, has_year_zero=has_year_zero)
+        # otherwise use a stripped-down version of C-python's _strptime.py
+        # (doesn't understand all possible formats, including locales and time zones)
```

Comment:
It will support the locale's month names and abbreviations, though, correct?  I think since you use the standard library `calendar` module for those [this will be the case](https://docs.python.org/3/library/calendar.html#calendar.month_name).

Reply:
yes, you are right.  I changed this to 
```
        # (doesn't understand all possible formats, just
        # 'y','Y','m','B','b','d','H','M','S' and 'f')
```